src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_device_connection_status_notification.cc

```diff
@@ -0,0 +1,171 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "sdl_rpc_plugin/commands/hmi/on_device_connection_status_notification.h"
+#include "policy/policy_table/enums.h"
+
+namespace sdl_rpc_plugin {
+using namespace application_manager;
+
+namespace commands {
+
+OnDeviceConnectionStatusNotification::OnDeviceConnectionStatusNotification(
+    const application_manager::commands::MessageSharedPtr& message,
+    ApplicationManager& application_manager,
+    rpc_service::RPCService& rpc_service,
+    HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handle)
+    : NotificationFromHMI(message,
+                          application_manager,
+                          rpc_service,
+                          hmi_capabilities,
+                          policy_handle) {}
+
+OnDeviceConnectionStatusNotification::~OnDeviceConnectionStatusNotification() {}
+
+void OnDeviceConnectionStatusNotification::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const smart_objects::SmartObject& msg_params =
+      (*message_)[strings::msg_params];
+  if (!msg_params.keyExists(strings::device)) {
+    LOG4CXX_WARN(logger_,
+                 "Invalid data: Key " << strings::device << " does not exist.");
+    return;
+  }
+  const smart_objects::SmartObject& device_info_so =
+      msg_params[strings::device];
+
+  std::string device_id;
+  hmi_apis::Common_UserSetting::eType usb_transport_status =
+      hmi_apis::Common_UserSetting::INVALID_ENUM;
+  if (!GetMandatoryParameters(
+          device_info_so, device_id, usb_transport_status)) {
+    LOG4CXX_WARN(logger_, "Error getting notification mandatory parameters");
+    return;
+  }
+
+  application_manager_.GetPolicyHandler().OnDeviceConnectionStatus(
+      device_id, usb_transport_status);
+}
+
+bool OnDeviceConnectionStatusNotification::GetMandatoryParameters(
+    const smart_objects::SmartObject& device_data,
+    std::string& out_device_id,
+    hmi_apis::Common_UserSetting::eType& out_usb_transport_status) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  int32_t transport_type;
+  int32_t usb_transport_status;
+
+  bool all_key_is_valid = true;
+  all_key_is_valid &=
+      GetKeyStringValue(device_data, strings::id, out_device_id);
```

Comment:
@ZhdanovP suggest to use local temp variable for output param here and in the end of the function, if everything is correct, assign to output variables local ones as it's done for `out_usb_transport_status`

Reply:
@AKalinich-Luxoft For define the out_device_id it's only one assign operation on line 131, I agree with your approach but think in this case it's redundant